File: assignments/assignment-1-solution/src-csharp/CrazyEights/CardDeck/DeckInitializer.cs
using CrazyEights.Cards;
using CrazyEights.Domain;

namespace CrazyEights.CardDeck;

/// <summary>
/// Creates and shuffles a new deck for the game.
/// </summary>
public static class DeckInitializer
{
    public static Deck CreateCardDeck(ProgramContext programContext)
    {
        var cards = CreateOrderedCardList();
        var shuffledCards = Shuffle(programContext, cards);
        return new Deck(shuffledCards);
    }

    private static List<ICard> CreateOrderedCardList()
    {
        var cards = new List<ICard>();
        foreach (var suit in Suit.GetSuits())
        foreach (var rank in Rank.GetRanks())
            cards.Add(new Card(rank, suit));

        return cards;
    }

    private static List<ICard> Shuffle(ProgramContext programContext, List<ICard> cards)
    {
        var random = programContext.RandomNumberGenerator;
        var shuffledCards = cards.OrderBy(c => random.Next()).ToList();
        return shuffledCards;
    }
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/carddeck/DeckInitializer.java
package edu.kennesaw.crazy8s.carddeck;

import edu.kennesaw.crazy8s.ProgramContext;
import edu.kennesaw.crazy8s.cards.Card;
import edu.kennesaw.crazy8s.cards.StandardCard;
import edu.kennesaw.crazy8s.domain.Rank;
import edu.kennesaw.crazy8s.domain.RankType;
import edu.kennesaw.crazy8s.domain.Suit;
import edu.kennesaw.crazy8s.domain.SuitType;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Creates and shuffles a new deck for the game.
 */
public final class DeckInitializer {
    private DeckInitializer() {
    }

    public static Deck createCardDeck(ProgramContext programContext) {
        List<Card> cards = createOrderedCardList();
        List<Card> shuffledCards = shuffle(programContext, cards);
        return new Deck(shuffledCards);
    }

    private static List<Card> createOrderedCardList() {
        List<Card> cards = new ArrayList<>();
        for (SuitType suit : Suit.getSuits()) {
            for (RankType rank : Rank.getRanks()) {
                cards.add(new StandardCard(rank, suit));
            }
        }

        return cards;
    }

    private static List<Card> shuffle(ProgramContext programContext, List<Card> cards) {
        List<Card> shuffledCards = new ArrayList<>(cards);
        Collections.shuffle(shuffledCards, programContext.getRandomNumberGenerator());
        return shuffledCards;
    }
}
